tests/unit/iotics/lib/identity/fake.py

```diff
@@ -23,4 +24,4 @@ def get_document(self, doc_id: str) -> RegisterDocument:
 
     def register_document(self, document: RegisterDocument, private_key: ec.EllipticCurvePrivateKey,
                           issuer: Issuer, audience: str = ''):
-        self.docs[issuer.did] = document
+        self.docs[issuer.did] = RegisterDocumentBuilder().build_from_dict(document.to_dict())
```

Comment:
Is the point here that it needs a copy?

Reply:
We get a copy here but also the document builder will do some validation giving an poor mans approximation of the validation done by the resolver